### PR TITLE
Fixes for mingw cross-compilation

### PR DESCRIPTION
**Pull request by tonytheodore**
Tested on both static/shared builds (untested runtime - no windows machine at hand).

`make test` on OSX native build has single failure `44 - test_many_sockets (Timeout)` both before and after changes.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -949,20 +961,18 @@ else ()
...
+      add_library (libzmq-static STATIC $<TARGET_OBJECTS:objects> ${public_headers} ${html-docs} ${readme-docs} ${zmq-pkgconfig} ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
+    endif ()
     set_target_properties (libzmq-static PROPERTIES
       PUBLIC_HEADER "${public_headers}"
       COMPILE_DEFINITIONS "ZMQ_STATIC"
       OUTPUT_NAME "zmq"
       PREFIX "lib")
   endif()
 
-  target_include_directories (objects
```
Comment (bluca) [inline CMakeLists.txt]: why is this being removed?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -192,6 +194,7 @@ foreach(test ${tests})
         link_directories(${test} PRIVATE "${CMAKE_SOURCE_DIR}/../lib")
     endif()
   endif()
+  set_target_properties(${test} PROPERTIES COMPILE_DEFINITIONS "${libzmq_DEFS}" )
```
Comment (sigiesec) [inline tests/CMakeLists.txt]: I don't think this is the adequate way to propagate compile definitions to dependent targets.

target_compiler_definitions(libzmq PUBLIC ...) should be used somewhere above instead, then they should be propagated automatically.

See https://cmake.org/cmake/help/v3.0/command/target_compile_definitions.html

File: unittests/CMakeLists.txt [hunk 3]
```diff
@@ -33,6 +34,7 @@ foreach(test ${unittests})
   
   target_link_libraries(${test} libzmq-static ${OPTIONAL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} unity)
   
+  set_target_properties(${test} PROPERTIES COMPILE_DEFINITIONS "${libzmq_DEFS}" )
```
Comment (sigiesec) [inline unittests/CMakeLists.txt]: See comment above

File: CMakeLists.txt (hunk 1 above)
Comment (tonytheodore) [reply CMakeLists.txt]: It's moved up to where the `add_library(objects ...` line is.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (tonytheodore) [reply tests/CMakeLists.txt]: Thanks, I'll have a look at `PUBLIC`.

File: CMakeLists.txt (hunk 1 above)
Comment (bluca) [reply CMakeLists.txt]: Ah sorry, missed that

File: tests/CMakeLists.txt (hunk 2 above)
Comment (tonytheodore) [reply tests/CMakeLists.txt]: Nice! `PUBLIC` is a very handy feature.

**Pull request merged**